src/class/cdc/cdc_device.c

```diff
@@ -416,7 +416,12 @@ bool cdcd_xfer_cb(uint8_t rhport, uint8_t ep_addr, xfer_result_t result, uint32_
   //       Though maybe the baudrate is not really important !!!
   if ( ep_addr == p_cdc->ep_in )
   {
-    if ( 0 == tud_cdc_n_write_flush(itf) )
+    uint32_t flushed = tud_cdc_n_write_flush(itf);
+
+    // invoke transmit callback to possibly refill tx fifo
+    if ( tud_cdc_tx_complete_cb ) tud_cdc_tx_complete_cb(itf);
+
+    if ( 0 == flushed && tu_fifo_empty(&p_cdc->tx_ff) )
```

Comment:
This is still not ideal since we may not transfer all available bytes e.g flush send out 5 bytes, but the complete_cb() fill more 200 bytes just a few statement later. Ideally it should be 

```
  if ( ep_addr == p_cdc->ep_in )
  {
    // invoke transmit callback to possibly refill tx fifo
    if ( tud_cdc_tx_complete_cb ) tud_cdc_tx_complete_cb(itf);

    if ( 0 == tud_cdc_n_write_flush(itf) )
    {
      // There is no data left, a ZLP should be sent if
      // xferred_bytes is multiple of EP size and not zero
      if ( xferred_bytes && (0 == (xferred_bytes % CFG_TUD_CDC_EP_BUFSIZE)) )
      {
        usbd_edpt_xfer(TUD_OPT_RHPORT, p_cdc->ep_in, NULL, 0);
      }
    }
  }
```